ntbs-service/Pages/Search/Index.cshtml.cs

```diff
@@ -93,7 +93,8 @@ public async Task<IActionResult> OnGetAsync(int? pageIndex)
             var count = notificationIdsAndCount.Item2;
             
             IEnumerable<Notification> notifications = await notificationService.GetNotificationsByIdAsync(notificationIds);
-            var notificationBannerModels = notifications.Select(NotificationBannerModel.WithLink);
+            var notificationBannerModels = notifications.Select(async n => NotificationBannerModel.WithLink(n, await authorizationService.CanEdit(User, n)))

```

Comment:
same complexity here - and repeated, perhaps there's some commonising to be achieved somehwere here 🤔